src/cmd-sign

```diff
@@ -174,28 +175,25 @@ def robosign_ostree(args, s3, build, gpgkey):
             raise Exception(msg)
         print(msg)
 
-        # ok, it's valid -- add it to the tarfile
+        # We've validated the commit, now re-export the repo
         ostree_image = build['images']['ostree']
-        commit_tarfile = os.path.join(builddir, ostree_image['path'])
-        commit_tarfile_new = os.path.join(d, ostree_image['path'])
-        subprocess.check_call(['cp-reflink', commit_tarfile,
-                               commit_tarfile_new])
-        os.chmod(commit_tarfile_new, 0o660)
-        with tarfile.open(commit_tarfile_new, 'a:') as t:
-            t.add(commitmeta_obj_path, arcname=commitmeta_obj_relpath)
-        ostree_image['size'] = os.path.getsize(commit_tarfile_new)
-        ostree_image['sha256'] = sha256sum_file(commit_tarfile_new)
-        # and the critical bits
-        shutil.copymode(commit_tarfile, commit_tarfile_new)
-        shutil.move(commit_tarfile_new, commit_tarfile)
+        exported_ostree_path = os.path.join(builddir, ostree_image['path'])
+        if exported_ostree_path.endswith('.ociarchive'):
+            subprocess.check_call(['rpm-ostree', 'ex-container', 'export', '--repo=tmp/repo', checksum, f'oci-archive:{exported_ostree_path}:latest'])
+        else:
+            tmp_tar = os.path.join(d, ostree_image['path'])
+            # To make things a bit more efficient, append the commitmeta at
+            # the end of the archive after reflinking.
+            subprocess.check_call(['cp-reflink', exported_ostree_path, tmp_tar])
+            with tarfile.open(tmp_tar, 'a:') as t:
+                t.add(metapath, arcname=f'objects/{checksum[:2]}/{checksum[2:]}.commitmeta')
+            shutil.move(tmp_tar, exported_ostree_path)
```

Comment:
Minor: would be nicer to have this be the last operation as before. (I.e. right before `build.write()`). Fine as is though since in practice a failure here just fails the whole build anyway.